### ISSUE TITLE
Bug: np.resize() causes invalid trigger generation and misleading watermark accuracy

### ISSUE THREAD

**Issue by FadingVortex** [shaoshuo-ss/FedTracker#1]
### Issue
Hi, thank you for your excellent work and open-sourcing the code!

I encountered a problem while reproducing the watermark accuracy experiments in the paper. I believe the root cause lies in the incorrect usage of np.resize() when generating trigger patterns in the `generate_waffle_pattern()` function File:[watermark.py](https://github.com/shaoshuo-ss/FedTracker/blob/main/watermark/watermark.py#L56) -- Line 56. 

### Problem Description
Currently, the trigger patterns are resized like this:
```python
pattern = np.resize(pattern, (args.image_size, args.image_size, args.num_channels))
```
However, `np.resize()` does not perform image resizing — it only reshapes the array by trimming or repeating data. When resizing a large image (e.g., 423×389×3) to CIFAR-10 size (32×32×3), this operation discards most of the original pixels, rather than scaling the image.

As a result, the spatial structure of the watermark pattern is broken or completely lost, leading to incorrect or meaningless trigger generation.

- The paper reports watermark accuracy as **high as 99%**, and similar results are seen in the released model.
- However, when I re-generated `trigger_set` using the existing function and tested watermark accuracy, it only achieved **~10%**, close to random guessing.
- I suspect this discrepancy is caused by **using the same** `trigger_set` **for both training and evaluation**, which overfits the distorted patterns during training with VGG16.

### Suggested Fix
I replaced the problematic resizing with a PIL-based method using interpolation:
```python
pattern = pattern.resize((args.image_size, args.image_size), Image.BILINEAR)
pattern = np.array(pattern)
```
After this change, the trigger set retains structural features, and watermark accuracy consistently returns to ~**99%**, as expected.

I suggest replacing np.resize() with PIL.Image.resize() followed by np.array() to ensure proper image scaling.

I’ve already implemented and tested the fix — happy to submit a PR if it helps!

Thanks again for your valuable work!

**Comment by shaoshuo-ss** [comment]
Dear FadingVortex:

Thank you for the suggestion! However, since this work has been finished for a long time, I may not have the energy and time to check and verify the correctness of the code and modifications in detail now. Therefore, I decided to add a note about this potential issue in the README to warn other readers.

Thank you again for the valuable suggestions!

**Comment by FadingVortex** [comment]
Dear Shuo Shao,

Wow! Thank you so much for taking the time to review the issue and for your kind reply. I truly appreciate your willingness to add a note in the README.

Wishing you all the best in your future research and endeavors!
